ansible/roles/providers/defaults/main.yml

```diff
@@ -0,0 +1,3 @@
+---
+packer_builder_type: ''
+provider_name: "{{ 'aws' if packer_builder_type.startswith('amazon') }}"
```

Comment:
packer_builder_type is defined automatically by packer, so we can use that to provide a default value, or override using the provider_name variable.